### PR TITLE
Fix of non-working examples

### LINKED ISSUE #345: Tiled Example broken?
I have been trying a couple of the examples and a lot of them don't work on their current form.

One example which I was really trying to make it work is the Tiled example. It does throw the following error stack:

```
E/flutter (23845): [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: ServicesBinding.defaultBinaryMessenger was accessed before the binding was initialized.
E/flutter (23845): If you're running an application and need to access the binary messenger before `runApp()` has been called (for example, during plugin initialization), then you need to explicitly call the `WidgetsFlutterBinding.ensureInitialized()` first.
E/flutter (23845): If you're running a test, you can call the `TestWidgetsFlutterBinding.ensureInitialized()` as the first line in your test's `main()` method to initialize the binding.
E/flutter (23845): #0      defaultBinaryMessenger.<anonymous closure> (package:flutter/src/services/binary_messenger.dart:76:7)
E/flutter (23845): #1      defaultBinaryMessenger (package:flutter/src/services/binary_messenger.dart:89:4)
E/flutter (23845): #2      PlatformAssetBundle.load (package:flutter/src/services/asset_bundle.dart:219:15)
E/flutter (23845): #3      Images._fetchToMemory (package:flame/images.dart:44:46)
E/flutter (23845): #4      Images.load (package:flame/images.dart:25:37)
E/flutter (23845): #5      main (package:littlemancomputer/main.dart:12:16)
E/flutter (23845): #6      _runMainZoned.<anonymous closure>.<anonymous closure> (dart:ui/hooks.dart:241:25)
E/flutter (23845): #7      _rootRun (dart:async/zone.dart:1184:13)
E/flutter (23845): #8      _CustomZone.run (dart:async/zone.dart:1077:19)
E/flutter (23845): #9      _runZoned (dart:async/zone.dart:1619:10)
E/flutter (23845): #10     runZonedGuarded (dart:async/zone.dart:1608:12)
E/flutter (23845): #11     _runMainZoned.<anonymous closure> (dart:ui/hooks.dart:233:5)
E/flutter (23845): #12     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:301:19)
E/flutter (23845): #13     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
```

From what I can see the problem is that the `TileGame` class tries to use the TiledComponent which works with Futures and that is where everything starts to come down...

I read somewhere that some recent changes to Flutter made some things break in Flame, however I thought that the examples on this repo should be up-to-date.

Can anyone help? If I do understand the problem I can happily fix the examples on the repo. Just my know-how of flame is not big enough for it.

Cheers

Some infos:

```
[✓] Flutter (Channel stable, v1.17.1, on Linux, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Android Studio (version 3.6)

Flame version: 0.21.0
```

### PR DESCRIPTION
**Pull request by Flavsditz**
# Description

As per my question on #345 I have added the `WidgetsFlutterBinding.ensureInitialized();` command to examples that didn't have it and were not working.

Fixes #345

## Type of change

- [ x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `develop`
- [x] This PR is targeted to merge into `develop` (not `master`)
- [ ] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [ ] The continuous integration (CI) is passing

### CODE REVIEW

File: doc/examples/README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 This folder contains a collection of more specific examples using flame, specially to showcase some more advanced functionality.
 
-To run this examples after cloning, first run `flutter create .` inside to generate the necessary files, as per instructed by the other flutter commands.
+To run this examples after cloning, first run `flutter create .` inside the desired example folder to generate the necessary files, as per instructed by the other flutter commands.
 
-After that, you should be able to `flutter run` just fine.
+After that, run `flutter pub get` to get all dependencies as defined in the pubspec.yml file.
+
+Finaly, you should be able to `flutter run` just fine.
```
Comment (erickzanardo) [inline doc/examples/README.md:9]: typo: `Finally`

Comment (Flavsditz) [reply doc/examples/README.md]: Let me fix that! I also found another problem with one of the examples which I fixed

File: doc/examples/README.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 This folder contains a collection of more specific examples using flame, specially to showcase some more advanced functionality.
 
-To run this examples after cloning, first run `flutter create .` inside to generate the necessary files, as per instructed by the other flutter commands.
+To run this examples after cloning, first run `flutter create .` inside the desired example folder to generate the necessary files, as per instructed by the other flutter commands.
```
Comment (spydon) [inline doc/examples/README.md:5]: Not your typo, but would be nice to get fixed in this PR anyways. :)
Should be "To run an example" or "these examples"

Comment (Flavsditz) [reply doc/examples/README.md]: Correct!

**Comment by erickzanardo** [comment]
Thanks @Flavsditz

File: doc/examples/README.md (hunk 2 above)
Comment (spydon) [reply doc/examples/README.md]: Thanks!

**Pull request merged**